src/client/Microsoft.Identity.Client/ManagedIdentity/ImdsV2ManagedIdentitySource.cs

```diff
@@ -0,0 +1,185 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.Http;
+using Microsoft.Identity.Client.Http.Retry;
+using Microsoft.Identity.Client.Internal;
+using Microsoft.Identity.Client.Utils;
+
+namespace Microsoft.Identity.Client.ManagedIdentity
+{
+    internal class ImdsV2ManagedIdentitySource : AbstractManagedIdentity
+    {
+        private const string CsrMetadataPath = "/metadata/identity/getPlatformMetadata";
+
+        private static CsrMetadata _csrMetadata;
```

Comment:
You cannot cache _csrMetadata because it varies based on the request context?

Reply:
@gladjohn - is the csr metadata cachable? if yes, by which key and for how long?

Reply:
I think we should not cache it. Just rely on token caching.

Reply:
per offline discussion, we will not cache it.